gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/JobId.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.bigquery;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.services.bigquery.model.JobReference;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Google BigQuery Job identity.
+ */
+public class JobId implements Serializable {
+
+  private static final long serialVersionUID = 1225914835379688976L;
+
+  private final String project;
+  private final String job;
+
+  /**
+   * Returns project's user-defined id
+   */
+  public String project() {
+    return project;
+  }
+
+  /**
+   * Returns the job's user-defined id.
+   */
+  public String job() {
+    return job;
+  }
+
+  private JobId(String project, String dataset) {
+    this.project = project;
+    this.job = dataset;
+  }
+
+  /**
+   * Creates a job identity given project's and job's user-defined id.
+   */
+  public static JobId of(String project, String job) {
+    return new JobId(checkNotNull(project), checkNotNull(job));
+  }
+
+  /**
+   * Creates a job identity given only its user-defined id.
+   */
+  public static JobId of(String job) {
+    return new JobId(null, checkNotNull(job));
```

Comment:
All identity objects (`DatasetId`, `TableId` and `JobId`) can be specified without providing a project's id. This is because a project's id is already set in `BigQueryOptions`. My idea is that we process all identity objects in the service layer and set the project's id to `options.projectId()` if it is `null`, before calling methods from the SPI layer.